### ISSUE TITLE
aws_iam_policy_document with a list of more than one element in identifiers, overrided with another aws_iam_policy_document crash. 

### ISSUE THREAD

**Issue by fbourqui** [terraform-providers/terraform-provider-aws#4085]
### Terraform Version
Terraform v0.11.6
+ provider.aws v1.13.0

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_iam_policy_document
- aws_iam_policy_document override previous

### Terraform Configuration Files
```hcl
data "aws_iam_policy_document" "test_bucket" {
  statement {
    sid = "AllowWritesWhileMaintainingOwnership"
    effect = "Allow"
    principals {
      type        = "AWS"
      # my real use case with a map workspace/account_alias to account_id
      #identifiers = ["${formatlist("arn:aws:iam::%s:role/test", values(var.map_acc_id))}"]
      # this always crash with a 2 element list and the following override, without the override it's fine
      identifiers = ["${formatlist("arn:aws:iam::%s:role/test", list("123456789012", "012345678901"))}"]
      # with no format list, it crash the same.
      # identifiers = ["${list("arn:aws:iam::123456789012:role/test", "arn:aws:iam::012345678901:role/test")}"]
      # previously erratic behaviour for the following testcase was due to remote file system kernel caching
      # this always work fine
      #identifiers = ["${formatlist("arn:aws:iam::%s:role/test", list("123456789012"))}"]
      # this always work fine
      #identifiers = ["${format("arn:aws:iam::%s:role/test", "123456789012")}"]
    }
    actions = ["s3:PutObject"]
    resources = ["arn:aws:s3:::test/*"]
    condition {
      test     = "StringEquals"
      variable = "s3:x-amz-acl"
      values   = ["bucket-owner-full-control"]
    }
  }
}

# This override is in a module in my real use case
data "aws_iam_policy_document" "policy" {
  override_json = "${data.aws_iam_policy_document.test_bucket.json}"
  # disable creation of public read or write object
  statement {
    sid = "ensure-private-read-write"
    effect = "Deny"
    principals {
      type        = "*"
      identifiers = ["*"]
    }
    actions = ["s3:PutObject",
              "s3:PutObjectAcl"]
    resources = ["arn:aws:s3:::test/*"]
    condition {
    test     = "StringEquals"
    variable = "s3:x-amz-acl"
    values   = ["public-read",
          "public-read-write"]
    }
  }
}
```

### Debug Output
https://gist.github.com/fbourqui/943cc0e52484ed20aa0ef1b9b2410bbb

### Panic Output
https://gist.github.com/fbourqui/943cc0e52484ed20aa0ef1b9b2410bbb

### Expected Behavior
Terraform should created the override policy without crashing.

### Actual Behavior
Terraform crash.

### Steps to Reproduce
1. `terraform apply` with the above code

### Important Factoids
Lot of different test run to find out it's the combination of a list with more than one element in identifiers and a policy override that create the crash. see all other example in comment and the outcome.

Without policy override all use-cases works as expected.

### References
Not to my knowledge.

**Comment by bflad** [comment]
The fix for this has been merged in via #4277 and will release with v1.16.0 of the AWS provider, likely mid next week.

**Comment by bflad** [comment]
This has been released in [version 1.16.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1160-april-25-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
